timemachine/cpp/src/kernels/k_exchange.cu

```diff
@@ -384,86 +420,117 @@ template void __global__ k_adjust_weights<double, 1>(
 template <typename RealType, int THREADS_PER_BLOCK>
 void __global__ k_set_sampled_weight_block(
     const int N,
+    const int batch_size,
     const int mol_size,
-    const int *__restrict__ target_atoms,
+    const int num_weights,
+    const int *__restrict__ target_atoms, // [batch_size + 1]
```

Comment:
Isn't this `batch_size*mol_size`? Not sure why we need the +1 either, since down below, we only ever access up to `target_atoms[(batch_size)*mol_size - 1]` elements

Reply:
Correct, the comment is wrong. Will fix

Reply:
Fixed in https://github.com/proteneer/timemachine/pull/1246/commits/d2dfbaa4c276477d50fe5e41d814dbb733567bb9 and renamed another `target_atoms` var that was incorrect